### ISSUE TITLE
Fix S3444 FN: Support static methods in interfaces

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6386]
### Description

FN for S3444 (InheritedCollidingInterfaceMembers).

Using C#11 static abstract/virtual methods in interfaces, the rule is not raising an issue.

### Repro steps

```csharp
    public interface IAbstractMethodFirst
    {
        public static abstract string StaticAbstractMethod(string value);
    }

    public interface IAbstractMethodSecond
    {
        public static abstract string StaticAbstractMethod(string value);
    }

    public interface IAbstractMethodCommon : IAbstractMethodFirst, IAbstractMethodSecond { } // FN

    public interface IVirtualMethodFirst
    {
        public static virtual string StaticVirtualMethod(string value) => value;
    }

    public interface IVirtualMethodSecond
    {
        public static virtual string StaticVirtualMethod(string value) => value;
    }

    public interface IVirtualMethodCommon : IVirtualMethodFirst, IVirtualMethodSecond { } // FN
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by gregory-paidis-sonarsource** [comment]
We decided to not only support static `abstract/virtual` methods, but static methods in general.
